hypershift-operator/controllers/platform/aws/controller.go

```diff
@@ -116,8 +150,16 @@ func (r *AWSEndpointServiceReconciler) Reconcile(ctx context.Context, req ctrl.R
 		}
 	}
 
-	// Reconcile the AWSEndpointService
-	if err = reconcileAWSEndpointService(ctx, awsEndpointService, r.ec2Client, r.elbv2Client); err != nil {
+	// Reconcile the AWSEndpointService Spec
+	if err := reconcileAWSEndpointServiceSpec(ctx, r, awsEndpointService); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to reconcile spec: %w", err)
+	}
+	if err := r.Update(ctx, awsEndpointService); err != nil {
```

Comment:
Shouldn't we gate this and only do the api call if we actually changed the spec?